edx_rbac/models.py

```diff
@@ -6,45 +6,77 @@
 from __future__ import absolute_import, unicode_literals
 
 # from django.db import models
+from django.contrib.auth.models import User
+from django.db import models
+from django.db.models.base import ModelBase
 from django.utils.encoding import python_2_unicode_compatible
 from model_utils.models import TimeStampedModel
+from django.db.models.fields import FieldDoesNotExist
+
+
+class UserRoleAssignmentCreator(ModelBase):
+    """
+    The model extending UserRoleAssignment should get a foreign key to a model that is a subclass of UserRole.
+    """
+
+    def __new__(cls, name, bases, attrs):
+        model = super(UserRoleAssignmentCreator, cls).__new__(cls, name, bases, attrs)
+        for b in bases:
+            if b.__name__ == 'UserRoleAssignment' and model.__name__ != b.__name__:
+                try:
+                    model._meta.get_field('role')
+                except FieldDoesNotExist:
+                    if model.role_class and issubclass(model.role_class, UserRole):
+                        model.add_to_class(
+                            'role',
+                            models.ForeignKey(model.role_class, db_index=True, on_delete=models.CASCADE),
+                        )
+                    else:
+                        raise Exception('role_class must be defined for any subclass of UserRole!')
+                return model
+        return model
 
 
 @python_2_unicode_compatible
 class UserRole(TimeStampedModel):
     """
-    TODO: replace with a brief description of the model.
-
-    TODO: Add either a negative or a positive PII annotation to the end of this docstring.  For more
-    information, see OEP-30:
-    https://open-edx-proposals.readthedocs.io/en/latest/oep-0030-arch-pii-markup-and-auditing.html
+    Model defining a role a user can have
     """
+    name = models.CharField(unique=True, max_length=255, db_index=True)
```

Comment:
In addition to a `name` for the role, we'll want a human/admin understood `description` of the role as well.